examples/fedpca_examples/dim_reduction/client.py

```diff
@@ -11,38 +11,45 @@
 from torch.utils.data import DataLoader
 from torchvision.transforms import transforms
 
-from examples.fedpca_examples.dim_reduction.mnist_model import MnistNet
+from examples.models.mnist_model import MnistNet
 from fl4health.clients.basic_client import BasicClient
-from fl4health.preprocessing.pca_preprocessor import PCAPreprocessor
-from fl4health.utils.dataset import MNISTDataset
+from fl4health.preprocessing.pca_preprocessor import PcaPreprocessor
+from fl4health.utils.dataset import BaseDataset, MnistDataset
 from fl4health.utils.metrics import Accuracy
+from fl4health.utils.random import set_all_random_seeds
 from fl4health.utils.sampler import DirichletLabelBasedSampler
 
 
-def get_mnist_dataset(data_dir: Path, train: bool) -> MNISTDataset:
+def get_mnist_dataset(data_dir: Path, train: bool) -> MnistDataset:
     transform = transforms.Compose(
         [
             transforms.ToTensor(),
             transforms.Normalize((0.5), (0.5)),
         ]
     )
-    return MNISTDataset(data_dir, train=train, transform=transform)
+    return MnistDataset(data_dir, train=train, transform=transform)
 
 
-class MnistFedPCAClient(BasicClient):
+class MnistFedPcaClient(BasicClient):
     def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
         batch_size = self.narrow_config_type(config, "batch_size", int)
         pca_path = Path(self.narrow_config_type(config, "pca_path", str))
         new_dimension = self.narrow_config_type(config, "new_dimension", int)
-        pca_preprocessor = PCAPreprocessor(pca_path)
+        pca_preprocessor = PcaPreprocessor(pca_path)
         sampler = DirichletLabelBasedSampler(list(range(10)), sample_percentage=0.6, beta=0.75)
 
-        train_loader = pca_preprocessor.reduce_dimension(
-            new_dimension, batch_size, True, sampler, get_mnist_dataset, self.data_path, True
-        )
-        val_loader = pca_preprocessor.reduce_dimension(
-            new_dimension, batch_size, False, sampler, get_mnist_dataset, self.data_path, True
-        )
+        # Get training and validation datasets.
+        train_dataset: BaseDataset = get_mnist_dataset(data_dir=self.data_path, train=True)
+        validation_dataset: BaseDataset = get_mnist_dataset(data_dir=self.data_path, train=False)
+        train_dataset = sampler.subsample(dataset=train_dataset)
```

Comment:
Maybe this is intentional, but it looks like the validation dataset is no longer being subsampled as it had been in the previous implementation.

Reply:
I'm not sure why I deleted the subsampling for validation data... It's been added back now